sycl/include/CL/sycl/detail/defines_elementary.hpp

```diff
@@ -118,3 +120,14 @@
 #define __SYCL_UNROLL(x)
 #endif // compiler switch
 #endif // __SYCL_UNROLL
+
+#if !defined(SYCL_DISABLE_CPP_VERSION_CHECK_MESSAGE) && __cplusplus < 201703L
+
+#if defined(_MSC_VER) && !defined(__clang__)
+__SYCL_WARNING("DPCPP does not support C++ version earlier than C++17. Some "
+               "features might not be available.")
+#else
+#warning: DPCPP does not support C++ version earlier than C++17. Some features might not be available.
```

Comment:
This is the only way to emit a warning from system header using clang, it cannot be wrapped by a macro(`__pragma` warning doesn't work in system headers). Borrowed solution from libcxx
.(TODO: Add this comment to the code in the next "address comment" patch).

The fact that `__SYCL_WARNING` does not work with clang should be handled separately.